hsconfig/config.go

```diff
@@ -71,28 +73,35 @@ func (c *Config) Save() error {
 }
 
 func (c *Config) AddToRecentProjects(filePath string) {
+	found := false
 	for idx := range c.RecentProjects {
 		if c.RecentProjects[idx] == filePath {
+			found = true
 			if idx != 0 {
 				old := c.RecentProjects[0]
 				c.RecentProjects[0] = filePath
 				c.RecentProjects[idx] = old
 			}
-			return
```

Comment:
This return was causing a bug where, if you loaded a project that was already in your recent projects list, you wouldn't hit the `c.Save()` below. This meant that your recent projects list would update during this session, but that would not get saved to the environment.json file for the next session.

Reply:
Good catch!